code/modules/admin/watchlist.dm

```diff
@@ -0,0 +1,118 @@
+client/proc/watchlist_add(target_ckey, browse = 0)
+	if(!target_ckey)
+		var/new_ckey = ckey(input(usr,"Who would you like to add to the watchlist?","Enter a ckey",null) as text)
+		if(!new_ckey)
+			return
+		new_ckey = sanitizeSQL(new_ckey)
+		var/DBQuery/query_watchfind = dbcon.NewQuery("SELECT ckey FROM [format_table_name("player")] WHERE ckey = '[new_ckey]'")
+		if(!query_watchfind.Execute())
+			var/err = query_watchfind.ErrorMsg()
+			log_game("SQL ERROR obtaining ckey from player table. Error : \[[err]\]\n")
+			return
+		if(!query_watchfind.NextRow())
+			usr << "<span class='redtext'>[new_ckey] has not been seen before, you can only add known players.</span>"
+			return
+		else
+			target_ckey = new_ckey
+	var/target_sql_ckey = sanitizeSQL(target_ckey)
+	if(check_watchlist(target_sql_ckey))
+		usr << "<span class='redtext'>[target_sql_ckey] is already on the watchlist.</span>"
+		return
+	var/reason = input(usr,"Please state the reason","Reason") as message|null
+	if(!reason)
+		return
+	reason = sanitizeSQL(reason)
+	var/timestamp = SQLtime()
+	var/adminckey = usr.ckey
+	if(!adminckey)
+		return
+	var/admin_sql_ckey = sanitizeSQL(adminckey)
+	var/DBQuery/query_watchadd = dbcon.NewQuery("INSERT INTO [format_table_name("watch")] (ckey, reason, adminckey, timestamp) VALUES ('[target_sql_ckey]', '[reason]', '[admin_sql_ckey]', '[timestamp]')")
+	if(!query_watchadd.Execute())
+		var/err = query_watchadd.ErrorMsg()
+		log_game("SQL ERROR during adding new watch entry. Error : \[[err]\]\n")
+		return
+	log_admin("[key_name(usr)] has added [target_ckey] to the watchlist - Reason: [reason]")
+	message_admins("[key_name_admin(usr)] has added [target_ckey] to the watchlist - Reason: [reason]", 1)
+	if(browse)
+		watchlist_show(target_sql_ckey)
+
+client/proc/watchlist_remove(target_ckey, browse = 0)
```

Comment:
Letting admins remove people from the watchlist is eh. It's kinda good, kinda bad.